netCDF4/_netCDF4.pyx

```diff
@@ -3822,6 +3822,13 @@ rename a `netCDF4.Variable` attribute named `oldname` to `newname`."""
             # else if variable has only add_offset attributes, rescale.
             elif hasattr(self, 'add_offset') and self.add_offset != 0.0:
                 data = data + self.add_offset
+
+        # if attribute _Unsigned is True, and variable has signed integer
+        # dtype, return view with corresponding unsigned dtype (issue #656)
+        is_unsigned = getattr(self, '_Unsigned', False)
+        if is_unsigned and data.dtype.str[1] == 'i':
```

Comment:
I don't think the `Variable.dtype` should be updated - since it should reflect the data type of the netcdf data stored in the file.  `Variable.dtype` plus `_Unsigned=True` imply that the returned data will be unsigned.  I agree that this might be problematic since the returned data now has a type different than what is in the file.  We already do this with data packed as short integers with `scale_factor` and `add_offset` though.